ASPNetCoreMasters/Controllers/ItemsController.cs

```diff
@@ -10,32 +10,57 @@
 
 namespace ASPNetCoreMasters.Controllers
 {
+    [ApiController]
+    [Route("[controller]")]
     public class ItemsController : ControllerBase
     {   
-        public int Get(int id)
+        [HttpGet]
+        public IActionResult GetAll()
         {
-            //var item = new ItemService();
-            //var result = item.GetAll(id);
+            return Ok("GetAll route");
+        }
 
-            return id;
+        [HttpGet("{itemId}")]
+        public IActionResult Get(int itemId)
+        {
+            var itemService = new ItemService();
+            itemService.GetItem(itemId);
+            return Ok($"Get {itemId}");
+        }
+
+        [HttpGet("GetByFilters")]
+        public IActionResult GetByFilters([FromQuery] Dictionary<string, string> filters)
+        {
+            return Ok($"GetByFilters");
         }
 
-        public IActionResult Post([FromBody] ItemCreateBindingModel itemCreateBidingModel)
+        [HttpPost("{itemcreatemodel}")]
```

Comment:
For model binding to occur there is no need to indicate this on the route. We only want primitive types such as int, double, etc if we want to use this approach. Kindly remove the route template and change to [HttpPost] or [HttpPost("")]